x-pack/elastic-agent/pkg/agent/transpiler/rules.go

```diff
@@ -1022,32 +1042,32 @@ func (r *MapRule) Apply(agentInfo AgentInfo, ast *AST) error {
 	)
 }
 
-func mapList(agentInfo AgentInfo, r *MapRule, l *List) error {
+func mapList(agentInfo AgentInfo, r *MapRule, l *List) (*List, error) {
 	values := l.Value().([]Node)
 
 	for idx, item := range values {
 		newAST := &AST{root: item}
 		for _, rule := range r.Rules {
 			err := rule.Apply(agentInfo, newAST)
 			if err != nil {
-				return err
+				return nil, err
 			}
 			values[idx] = newAST.root
 		}
 	}
-	return nil
+	return l, nil
```

Comment:
returning same what we passed in?

Reply:
This is so `mapList` and `mapDict` have the same behavior of returning possible a new node. In the case of `mapList` it will not return a new node, but in `mapDict` it does.